onchaind/onchain.c

```diff
@@ -502,8 +502,55 @@ static void handle_htlc_onchain_fulfill(struct tracked_output *out,
 	status_failed(STATUS_FAIL_INTERNAL_ERROR, "FIXME: %s", __func__);
 }
 
+static void resolve_htlc_tx(struct tracked_output ***outs,
+			    size_t out_index,
+			    const struct bitcoin_tx *htlc_tx,
+			    const struct sha256_double *htlc_txid,
+			    u32 tx_blockheight)
+{
+	struct tracked_output *out;
+	struct bitcoin_tx *tx;
+	u8 *wscript = bitcoin_wscript_htlc_tx(htlc_tx, to_self_delay[LOCAL],
+					      &keyset->self_revocation_key,
+					      &keyset->self_delayed_payment_key);
+
+
+	/* BOLT #5:
+	 *
+	 * A node SHOULD resolve its own HTLC transaction output by spending
+	 * it to a convenient address.  A node MUST wait until the
+	 * `OP_CHECKSEQUENCEVERIFY` delay has passed (as specified by the
+	 * other node's `open_channel` `to_self_delay` field) before spending
+	 * the output.
+	 */
+	out = new_tracked_output(outs, htlc_txid, tx_blockheight,
```

Comment:
We'll probably eventually want to add these to the DB as well so we don't assume that `onchaind` stays up the entire time. So we end up with two launch modes for `onchaind`: a hurry and settle on chain, and a go gather HTLCs mode. I'm guessing the HTLC state should already be sufficient to reconstruct the gathering transactions?